docs/fundamentals/code-analysis/style-rules/ide0130.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "IDE0130: Namespace does not match folder structure"
+description: "Learn about code analysis rule IDE0130: Namespace does not match folder structure"
+ms.date: 12/05/2022
+ms.topic: reference
+f1_keywords:
+- IDE0130
+helpviewer_keywords:
+- IDE0130
+author: gewarren
+ms.author: gewarren
+dev_langs:
+- CSharp
+- VB
+---
+# Namespace does not match folder structure (IDE0130)
+
+| Property                 | Value                                                  |
+| ------------------------ | ------------------------------------------------------ |
+| **Rule ID**              | IDE0130                                                |
+| **Title**                | Namespace does not match folder structure              |
+| **Category**             | Style                                                  |
+| **Subcategory**          | Language rules (Namespace naming preferences)          |
+| **Applicable languages** | C# and Visual Basic                                    |
+| **Options**              | `dotnet_style_namespace_match_folder`                  |
+
+## Overview
+
+This style rule uses the folder structure of the project to enforce namespace naming requirements.
+
+## Options
+
+Options specify the behavior that you want the rule to enforce. For information about configuring options, see [Option format](language-rules.md#option-format).
+
+### dotnet_style_namespace_match_folder
+
+| Property                 | Value                                                | Description                                        |
+| ------------------------ | ---------------------------------------------------- | -------------------------------------------------- |
+| **Option name**          | dotnet_style_namespace_match_folder                  |                                                    |
+| **Option values**        | `true`                                               | Prefer namespace naming to match folder structure. |
+|                          | `false`                                              | Disables the rule.                                 |
+| **Default option value** | `false`                                              |                                                    |
+
+## Example
+
+Assume that the following code snippets are from a file named `Data/Example.cs`, where `Data` represents the folder structure from the project file. The folder structure naming is added to the root namespace, which in this example is `Root`.
+
+### C\#
+
```

Comment:
```suggestion
```

Reply:
Do we not want headers here? Just a styling question, I'm happy either way

Reply:
@ryzngard The site will show a selector in the page and show only one of the snippets if they are put one after another without headers in between. For example:

![image](https://user-images.githubusercontent.com/31348972/206022233-4b366a4c-1068-48ea-b8ba-7e209a4dd43c.png)

This is what `dev_langs` is for.